### DISCUSSION TITLE
Remove restart flag when restart is canceled and exit is pressed instead

### DISCUSSION THREAD

**Discussion by LittleEndu** [ppy/osu#21003] [Q&A]
I pressed the notification for "update is ready" but then chose to continue playing the game when the exit prompt appeared. When you then exit the game, restart still occurs.

**Comment by peppy** [comment]
Can you show which prompt you saw in this case?

**Reply by LittleEndu** [reply]
The "are you sure you want to exit" prompt. I'm not at PC right now so can't take a screenshot (and mobile doesn't have that prompt)

**Reply by LittleEndu** [reply]
What I see:

https://user-images.githubusercontent.com/11320483/199299684-2e5a479f-9422-4d7a-bef3-a037f6d3a27d.mp4

**Comment by TheShadowEevee** [comment]
I took a video of this happening when quitting the game via the normal exit button and using the escape key. Forgive the terrible recording, my laptop was not happy about Wallpaper Engine, OBS, and osu all running at the same time.

https://user-images.githubusercontent.com/32500606/199293065-3ec9823a-0503-44b4-92b6-13907cf43f8d.mp4

**Reply by TheShadowEevee** [reply]
I used version 2022.1031.0 as the old version, shown with the installer I ran from my Desktop in the video, and let it auto-update to todays 2022.1101.0 version to show the restart happening

**Comment by TheShadowEevee** [comment]
The reason this happens appears to be because when the update notification is selected, [`restartToApplyUpdate()`](https://github.com/ppy/osu/blob/f1138f51f32c16f8b76bf1054239c6ed6ce330a8/osu.Desktop/Updater/SquirrelUpdateManager.cs#L142) is called, which then calls `Squirrel.UpdateManager.RestartAppWhenExited()` before calling `game.AttemptExit()` which can be cancelled. 

I'm unsure if it's possible to cancel Squirrel's restart call, as that's how it updates. I think the best "solution" would be to just exit, instead of using `AttemptExit()` which asks the user to confirm, since they already clicked the restart notification. That doesn't feel like the best route since it takes away that last chance to backtrack and going around that exit function is probably avoided elsewhere.

**Reply by CxVercility** [reply]
I was just about to submit this actually.
If someone clicks the notification to restart, there shouldn't be an additional dialogue that asks you whether you *really* want to exit the game. Just do it.

**Comment by peppy** [comment]
https://github.com/ppy/osu/pull/21295
